libraries/AP_RCTelemetry/AP_CRSF_Telem.cpp

```diff
@@ -389,7 +389,11 @@ bool AP_CRSF_Telem::is_packet_ready(uint8_t idx, bool queue_empty)
     case PARAMETERS:
         return _pending_request.frame_type > 0;
     case VTX_PARAMETERS:
+#if AP_VIDEOTX_ENABLED
```

Comment:
I'm not really convinced this is a useful thing to do. This assumes you for some reason want to build CRSF telemetry without VTX support. A more useful thing to do would be to make yaapu telemetry optional since that is much bigger flash-wise - but no-one wants to do this anyway. I don't think you will want this for AP_Periph and it just ugifies the code.

Reply:
Why is building with CRSF telemetry without VTX at all weird?  Neither of my Crossifre-using vehicles has a video transmitter on them.  And I know others in the same boat.

We'd be adding unnecessary coupling between CRSF and the VTX library if we were to make that technical dependency of one on the other.

I didn't actually do it for my daily flyer, but I did just check and that board's surprisingly close to the edge.  Excising videotx removes an additional 10% free space on that board (~+3kB).  Given that both tramp and smartaudio are minimize-features, it seems saving an additional 3kB on minimized boards by adding AP_VIDEOTX_ENABLED is a no-brainer - but I'd like to have the option of CRSF telem on there!

Reply:
I use CRSF....and both HD and analog video..But not smart audio video..on all my vehicles

Reply:
I would be happy to option out FRSKY D and only have passthru as standard....its obsolete

Reply:
> I would be happy to option out FRSKY D and only have passthru as standard....its obsolete

We already have options to do that one :-)

Reply:
> @peterbarker the reason I am reluctant is its more testing for me on CRSF. I am not objecting to making video tx optional - just objecting to the small slicing within the CRSF library itself. How much does this actually save, much less than 3k I 
imagine.

It saves an *additional* 3916 bytes on this branch.  That's on top of the ~4500 bytes saved from not compiling in tramp and smartaudio.

```
 everything: bin/arducopter   1813892      3500   258876               1817392          148672
no backends: bin/arducopter   1809388      3500   258876               1812888          153176
 no videotx: bin/arducopter   1805472      3368   259000               1808840          157228
```

>  Do you use yaapu? If you don't there is a mountain of flash to be saved there.

Yes, I use yaapu telemetry.  As does @Hwurzburg , pretty sure.  But this isn't an and/or thing - we can do both.  In fact, I think the bulk of yaapu's stuff might be behind `AP_FRSKY_SPORT_PASSTHROUGH_ENABLED`

Reply:
> > I would be happy to option out FRSKY D and only have passthru as standard....its obsolete
> 
> We already have options to do that one :-)

what I meant was that it could be excluded from the standard builds now with only passthru used